client/ayon_core/plugins/publish/collect_anatomy_instance_data.py

```diff
@@ -352,6 +352,17 @@ def fill_anatomy_data(self, context):
             if resolution_height:
                 anatomy_data["resolution_height"] = resolution_height
 
+            # make render layer name (or ROP name) available in the
+            # templates
+            render_layer = instance.data.get("renderlayer")
+            if render_layer:
+                anatomy_data["renderlayer"] = render_layer
+
+            # make AOV name if present available for templates
+            aov = instance.data.get("aov")
+            if aov:
+                anatomy_data["aov"] = aov
```

Comment:
What's "renderlayer" if not part of the product name already? This seems like an odd decision to allow `renderlayer` (which as far as I know doesn't really have a valid connotation in Houdini?) to be part of the anatomy template? Or?

Reply:
you might want to have render layer as a part of the path (not the whole product name).

Reply:
> you might want to have render layer as a part of the path (not the whole product name).

When doing so - what would you anticipate the product name be instead? 

Coming back to this from your PR description:

> This was bypassing any template set in `ayon+settings://core/tools/creator/product_name_profiles`. With this change, one can set template like this:
> 
> `{product[type]}<{Renderlayer}><_{Aov}>`
> 
> resulting in:
> 
> `renderMainlayer_normal`
> 
> It is up to TD to set correctly such template to make sure, product name will be stable enough. As a side-effect, optional keys can be used now in product name templates adding flexibility (and perhaps some danger of misuse).
> 
> `renderlayer` is now set on instance along with `aov` and is collected to anatomy data, so it can be used in template formatting so you can use template for renders like:

In practice a user should usually never see this. They see the product name or representation in the loader or the manager. Which raises the question - how come its _this_ crucial in the published filepath which usually isn't something that is displayed to the artists anyway?

Reply:
I think it is just an option to categorize things when you need to have access to them outside loaders/pipeline. We get those requests from time to time when you have part that is not integrated and you need to access the files directly. Then dropping the whole render layer folder to DCC with all of it's AOV sequences can speed things up.

Reply:
You mean to "force" it to context on representation even without it being used in the template so it can be picked up later on by delivery action? Like we are storing `udim` here https://github.com/ynput/ayon-core/blob/4d84ed8696657d976b6939ec1afab044e943818f/client/ayon_core/plugins/publish/integrate.py#L856-L859 (and also `frame` somewhere)

Reply:
Yes, similar to that. I'd expect this to way more valuable for delivery templates than production templates since in production templates it may be likely you will have duplicity anyway since usually the AOV is part of the product name as well.